packages/x-charts-pro/src/ChartZoomSlider/internals/ChartAxisZoomSlider.tsx

```diff
@@ -141,6 +147,140 @@ export function ChartAxisZoomSlider({ axisDirection, axisId }: ChartZoomSliderPr
   );
 }
 
+interface ChartAxisZoomSliderTrackProps extends React.ComponentProps<'rect'> {
+  axisId: AxisId;
+  axisDirection: 'x' | 'y';
+  reverse: boolean;
+}
+
+function ChartAxisZoomSliderTrack({
+  axisId,
+  axisDirection,
+  reverse,
+  ...other
+}: ChartAxisZoomSliderTrackProps) {
+  const ref = React.useRef<SVGRectElement>(null);
+  const { instance, svgRef } = useChartContext<[UseChartProZoomSignature]>();
+  const store = useStore<[UseChartProZoomSignature]>();
+
+  const onPointerDown = function onPointerDown(event: React.PointerEvent<SVGRectElement>) {
+    const rect = ref.current;
+    const element = svgRef.current;
+
+    if (!rect || !element) {
+      return;
+    }
+
+    const pointerDownPoint = getSVGPoint(element, event);
+    let zoomFromPointerDown = calculateZoomFromPoint(store.getSnapshot(), axisId, pointerDownPoint);
+
+    if (zoomFromPointerDown === null) {
+      return;
+    }
+
+    const { minStart, maxEnd } = selectorChartAxisZoomOptionsLookup(store.getSnapshot(), axisId);
+
+    // Ensure the zoomFromPointerDown is within the min and max range
+    zoomFromPointerDown = Math.max(Math.min(zoomFromPointerDown, maxEnd), minStart);
+
+    let pointerMoved = false;
+
+    const onPointerMove = rafThrottle(function onPointerMove(pointerMoveEvent: PointerEvent) {
```

Comment:
It's a bit weird to define the `onPointerMove` inside the `onPointerDown`

Does removing the `pointerMove`/`pointerDown` from `document` makes a difference in terms of performances?

Reply:
> It's a bit weird to define the onPointerMove inside the onPointerDown

Why? I think it makes sense. We only need to listen to the pointer move events after a pointer down happens, so it makes sense that we only add the listener after a pointer down event. 

We could add the event listener regardless, but then we're calling a function on every move just to do nothing. Seems a bit useless, IMO. 

> Does removing the pointerMove/pointerDown from document makes a difference in terms of performances?

It's probably negligible, I didn't test it.